iceoryx2-ffi/cxx/include/iox2/node_name.hpp

```diff
@@ -15,16 +15,16 @@
 
 #include "internal/iceoryx2.hpp"
 #include "iox/expected.hpp"
+#include "iox/string.hpp"
+#include "iox2/iceoryx2_settings.hpp"
 #include "semantic_string.hpp"
 
-#include <string>
-
 namespace iox2 {
 class NodeName {
   public:
     static auto create(const char* value) -> iox::expected<NodeName, SemanticStringError>;
 
-    auto as_string() const -> const std::string&;
+    auto to_string() const -> iox::string<NODE_NAME_LENGHT>;
```

Comment:
Not in this PR but as food for thought. I think a way to get to the string without copying would be nice, i.e. a `c_str` method.